chain/consensus/hierarchical/actors/subnet/genesis.go

```diff
@@ -121,10 +122,19 @@ func WriteGenesis(
 		if err != nil {
 			return err
 		}
-		b, err = makeTendermintGenesisBlock(context.TODO(), bs, *template, checkPeriod)
+		b, err = makeTendermintGenesisBlock(ctx, bs, *template, checkPeriod)
 		if err != nil {
 			return xerrors.Errorf("error making genesis tendermint block: %w", err)
 		}
+	case hierarchical.MirBFT:
```

Comment:
Let's make it an issue, but I am starting to feel that genesis generation is common for all consensus and we can probably extract it to a common function and simplify this code.

Reply:
Thinking it through, I don't think it'll be a lot of work, in case you want to include this already in this PR.

Reply:
The code is different for delegated. I created an issue https://github.com/filecoin-project/eudico/issues/187

Reply:
LGTM. Closing the issue.